data_model/src/isi.rs

```diff
@@ -757,50 +757,63 @@ pub mod error {
+            #[display(fmt = "Evaluation failed: {_0}")]
+            Evaluate(#[cfg_attr(feature = "std", source)] EvaluationError),
+            /// Failure to assert logical invariants in the sytem
+            #[display(fmt = "Validation failed: {_0}")]
+            Validate(#[cfg_attr(feature = "std", source)] ValidationError),
+            /// Query Error
+            #[display(fmt = "Query failed. {_0}")]
+            Query(#[cfg_attr(feature = "std", source)] QueryExecutionFailure),
+            /// Conversion Error
+            #[display(fmt = "Conversion Error: {_0}")]
+            Conversion(#[skip_from] #[skip_try_from] String),
             /// Failed to find some entity
-            #[display(fmt = "Failed to find. {_0}")]
+            #[display(fmt = "Entity missing: {_0}")]
             Find(#[cfg_attr(feature = "std", source)] Box<FindError>),
-            /// Failed to assert type
-            #[display(fmt = "Type assertion failed. {_0}")]
-            Type(#[cfg_attr(feature = "std", source)] TypeError),
+            /// Repeated instruction
+            #[display(fmt = "Repetition")]
+            Repetition(InstructionType, IdBox),
             /// Failed to assert mintability
-            #[display(fmt = "Mintability violation. {_0}")]
+            #[display(fmt = "{_0}")]
             Mintability(#[cfg_attr(feature = "std", source)] MintabilityError),
             /// Failed due to math exception
-            #[display(fmt = "Math error. {_0}")]
+            #[display(fmt = "Illegal math operation: {_0}")]
             Math(#[cfg_attr(feature = "std", source)] MathError),
-            /// Query Error
-            #[display(fmt = "Query failed. {_0}")]
-            Query(#[cfg_attr(feature = "std", source)] QueryExecutionFailure),
             /// Metadata Error.
             #[display(fmt = "Metadata error: {_0}")]
             Metadata(#[cfg_attr(feature = "std", source)] metadata::Error),
-            /// Unsupported instruction.
-            #[display(fmt = "Unsupported {_0} instruction")]
-            Unsupported(InstructionType),
             /// [`FailBox`] error
-            #[display(fmt = "Execution failed {_0}")]
+            #[display(fmt = "Execution failed: {_0}")]
             FailBox(#[skip_from] #[skip_try_from] String),
-            /// Conversion Error
-            #[display(fmt = "Conversion Error: {_0}")]
-            Conversion(#[skip_from] #[skip_try_from] String),
-            /// Repeated instruction
-            #[display(fmt = "Repetition")]
-            Repetition(InstructionType, IdBox),
-            /// Failed to validate.
-            #[display(fmt = "Failed to validate: {_0}")]
-            Validate(#[cfg_attr(feature = "std", source)] ValidationError),
+        }
+
+        /// Type error
+        #[derive(Debug, Display, Clone, PartialEq, Eq, FromVariant)]
+        // TODO: Only temporarily opaque because of problems with FFI
+        #[ffi_type(opaque)]
+        pub enum EvaluationError {
```

Comment:
Okay, I see a comment about type here. Maybe rename it somehow? I.e. `TypeEvaluationError`?

Reply:
I'm closing this comment in favor of the one before